docs: update realtime websockets guide (#19660)
directus/directus: docs/guides/real-time/chat/javascript.md

Inside of the `<script>`, create a `url` variable being sure to replace `your-directus-url` with your project’s URL:

```js
const url = 'https://your-directus-url';
```

## Import the SDK Composables

At the top of the `<script>` tag, import the SDK composables needed for this project

```html
<!doctype html>
<html>
	<body>
		<script>
			import { createDirectus, authentication, realtime } from 'https://www.unpkg.com/@directus/sdk/dist/index.js'; // [!code ++]

			const url = 'https://your-directus-url';
		</script>
	</body>
</html>
```

> - `createDirectus` is the hook that initializes a Directus client.

Reviewer: ```suggestion
- `createDirectus` is a function that initializes a Directus client.
```

Written like a true react dev hehe

> - `authentication` authenticates a user with the login and password.

Reviewer: ```suggestion
- `authentication` provides methods to authenticate a user.
```

Reviewer: Apply consistently across guides

> - `realtime` establishes a WebSocket connectivity.

Reviewer: ```suggestion
- `realtime` provides methods to establish a WebSocket connection.
```

Reviewer: Apply consistently across guides

});
```

Within the `#login` form submit event handler, get access to the email and password values

```js
const email = event.target.elements.email.value;
const password = event.target.elements.password.value;
```

Now, call the login method on the client, passing the email and password

```js
document.querySelector('#login').addEventListener('submit', function (event) {
	event.preventDefault();

  const email = event.target.elements.email.value;
  const password = event.target.elements.password.value;

  client.login(email, password); // [!code ++]
});
```

> Once the client is authenticated, immediately create a WebSocket connection:

Reviewer: Immediately works too, because that's what's happening in this example: auth then immediately connect. I do understand your point though, in that it sounds like you must immediately connect and agree with your suggestion. 

The key difference is that this is a project-based guide so it can be more instructional.

> Once the client is authenticated, immediately create a WebSocket connection:

Author: I get what you mean, Tim but I agree with leaving it as immediately since this is an instructional guide.

```js
client.connect();
```

## Subscribe To Messages

As soon as you have successfully authenticated, a message will be sent. When this happens, subscribe to updates on the
`Messages` collection.

```js
client.onWebSocket('message', function (message) {
  if (data.type == 'auth' && data.status == 'ok') {

> 	  subscribe('update')

Reviewer: This isn't a complex enough guide to leave 'unresolved threads' like above - where you call a function and then write it. For this reason, I'd suggest flipping the order - define function, and then call it.

Author: I get what you mean, but it made sense to do it this way because of the sentence that preceded talking about what to do after authentication.

  }
});
```

Create a `subscribe` function that subscribes to events.

```js
async function subscribe(event) {
  const { subscription } = await client.subscribe('messages', {
    event,
    query: {
      fields: ['*', 'user_created.first_name'],
    },

>   });

Reviewer: Yeah but isn't strictly needed for this project so I wouldn't. They're a tool that can be used, or not.

>   });

Reviewer 2: True it's not explicitly required here but still valuable if people will be using this as a starting off point potentially adding more than one `subscription` or `items request`. But feel free to resolve the suggestion

  for await (const message of subscription) {
    receiveMessage(message);
  }
}
```

When a subscription is started, a message will be sent to confirm. Create a `receiveMessage` function with the
following:

```js

> if (data.type == 'subscription' && data.event == 'init') {

Reviewer: https://github.com/directus/directus/pull/19660#discussion_r1448712726

	event.preventDefault();
	const text = event.target.elements.text.value; // [!code ++]

  client.sendMessage({ // [!code ++]
    type: 'items', // [!code ++]
    collection: 'messages', // [!code ++]
    action: 'create', // [!code ++]

>     data: { text}, // [!code ++]

Reviewer 2: ```suggestion
    data: { text }, // [!code ++]
```

## Display Historical Messages

To display the list of all existing messages, create a function `readAllMessages` with the following:

```js
function readAllMessages() {
  client.sendMessage({
    type: 'items',
    collection: 'messages',
    action: 'read',
    query: {
      limit: 10,
      sort: '-date_created',
      fields: ['*', 'user_created.first_name'],
    },
  });
}
```

Invoke this function directly before subscribing to any events

```js
client.onWebSocket('message', function (data) {
  if (data.type == 'auth' && data.status == 'ok') {

>     readAllMessages() // [!code ++]

Reviewer: ```suggestion
    readAllMessages(); // [!code ++]
```

Reviewer: Sweep and apply to all instances of running this function across all guides

		</form>

		<script>
		  import {
        createDirectus,
        authentication,
        realtime,

>       } from 'https://www.unpkg.com/@directus/sdk/dist/index.js';

Reviewer 2: Indentation has gone a bit wonky here in the full example.

Reviewer: Or, rather, is inconsistent. I prefer the one-liner for this.
